state/execution.go

```diff
@@ -339,19 +340,22 @@ func updateValidators(currentSet *types.ValidatorSet, abciUpdates []abci.Validat
 			// remove val
 			_, removed := currentSet.Remove(address)
 			if !removed {
-				return fmt.Errorf("Failed to remove validator %X", address)
+				return fmt.Errorf("failed to remove validator %X", address)
 			}
 		} else if val == nil {
-			// add val
+			// Set Accum to -totalVotingPower to make sure validators can't unbond/rebond to reset their (potentially
+			// previously negative) Accum to zero:
+			valUpdate.Accum = -totalVotingPower
+			// add validator
```

Comment:
also not needed

Reply:
That comment existed before already (as "// add val").  Similar comments exist for "remove val / update val". Either we should remove all of them (or keep all of them).

Reply:
The placing is important (before it was right after `if`)